src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1101,6 +1101,18 @@ function ($a) {
                                         ->prototype('variable')
                                         ->end()
                                     ->end()
+                                    ->arrayNode('retry_strategy')
+                                        ->addDefaultsIfNotSet()
+                                        ->validate()
+                                            ->ifTrue(function ($v) { return 'messenger.retry.exponential_retry_strategy' !== $v['service'] && (isset($v['max_retries']) || isset($v['exponential_offset'])); })
+                                            ->thenInvalid('"service" be used along with either "max_retries" or "exponential_offset".')
+                                        ->end()
+                                        ->children()
+                                            ->scalarNode('service')->defaultValue('messenger.retry.exponential_retry_strategy')->end()
+                                            ->scalarNode('max_retries')->defaultValue(3)->end()
+                                            ->scalarNode('exponential_offset')->defaultValue(0)->end()
```

Comment:
Here also we should validate that `exponential_offset` is numeric (and >= 0 ?).

Reply:
If we keep this option (still checking into if this is the best way to make this all configurable), `exponential_offset` CAN be negative, actually.